src/uu/split/src/split.rs

```diff
@@ -288,6 +309,134 @@ fn num_prefix(i: usize, width: usize) -> String {
     c
 }
 
+#[cfg(unix)]
+/// A writer that writes to a shell_process' stdin
+struct FilterWriter {
+    /// Running shell process
+    shell_process: Child,
+}
+
+#[cfg(unix)]
+impl Write for FilterWriter {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        self.shell_process
```

Comment:
could you add a comment why we are using a shell here?

Reply:
will add it in `struct FilterWriter` 's docstring